### PR TITLE
Add media scss mixin

### PR DESCRIPTION
**Pull request by alexander-schranz**
Configuration:

```scss
$breakpoints: (
    desktop: 1199px,
    tablet: 991px,
    smart: 767px,
    mobile: 577px,
);
```

**Example usage `media`**:

Inspired by: https://css-tricks.com/approaches-media-queries-sass/

```scss
@includa media('<tablet') {
    color: red;
}
```

Output:

```css
@media only screen and (max-width: 767px) {
    color: red;
}
```

**Example usage `mediaEachMax`**:

Make it easer to set attributes foreach media query:

```scss
$container-spaces: (
    default: 20px;
    desktop: 62px,
    tablet: 64px,
    smart: 30px,
);

.container {
    @include mediaEachMax($container-spaces, (
        margin-left,
        margin-right
    ));
}
```

Output:

```css
.container {
  margin-left: 20px;
  margin-right: 20px;
}

@media only screen and (max-width: 1199px) {
  .container {
    margin-left: 62px;
    margin-right: 62px;
  }
}

@media only screen and (max-width: 991px) {
  .container {
    margin-left: 64px;
    margin-right: 64px;
  }
}

@media only screen and (max-width: 767px) {
  .container {
    margin-left: 30px;
    margin-right: 30px;
  }
}
```

### CODE REVIEW

File: scss/tools/media.scss [hunk 1]
```diff
@@ -0,0 +1,110 @@
+// Media mixins
+$media-breakpoints: $breakpoints !default;
+
+// Media min automatically create a "only screen and ( min-width: $param )" media query.
+@mixin mediaMin($pixels) {
+    @media only screen and ( min-width: $pixels ) {
```
Comment (c00n84) [inline scss/tools/media.scss]: @alexander-schranz remove spacing and check the whole file.

**Comment by c00n84** [comment]
else ACK

File: scss/tools/media.scss (hunk 1 above)
Comment (alexander-schranz) [reply scss/tools/media.scss]: @c00n hab dies zu stylelint hinzugefügt ebenfalls musste ich die if else hinzufügen: https://github.com/massiveart/stylelint-config-ma/pull/2

**Comment by alexander-schranz** [comment]
needs: https://github.com/massiveart/stylelint-config-ma/pull/2 to complete the CI

File: scss/tools/_media.scss [hunk 2]
```diff
@@ -0,0 +1,111 @@
+// Media mixins
+$media-breakpoints: $breakpoints !default;
+
+// Media min automatically create a "only screen and ( min-width: $param )" media query.
```
Comment (thomasduenser) [inline scss/tools/_media.scss]: Remove whitespace in in `()` also in comments.

**Comment by thomasduenser** [comment]
@alexander-schranz else ACK

**Pull request merged**